### ISSUE TITLE
DFU upload problem on win32 - DFU ABORT bug

### ISSUE THREAD

**Issue by mbolivar** [leaflabs/libmaple#9]
See http://forums.leaflabs.com/topic.php?id=128

TLDR: DFU is not ABORTING upload transactions after a failed packet; it should.  Why packets are failing is a separate issue, which can be fixed with better DFU error handling. 

On some machines, there is a high DFU failure rate, from which Maple does not gracefully recover. During these botched uploads, Maple gets stuck in the DFU state machine somewhere, and does not accept attempts to retry DFU.  The root cause of this is very likely due to some problem with how USB is buffering data.  Perhaps some data is being lost and breaking the state machine.

Possible workarounds: reducing the transfer speed, reducing the DFU packet size.
